app/controllers/shops_controller.rb

```diff
@@ -33,14 +32,16 @@ def set_user
 
   def set_shop 
     @shop = @user.shop
+    authorize @shop
   end
 
   def shop_params 
     params.require(:shop).permit(:name, :description)
   end
 
   def check_shop_exist
-    if current_user.shop.present?
+    authorize Shop.new
```

Comment:
why do we authorize new shop here?

Reply:
This is my mistake, and I realize that we don't need to authorize anything there, so I delete that line.